src/Features/CSharp/Portable/Completion/CompletionProviders/OperatorIndexerCompletionProvider.cs

```diff
@@ -0,0 +1,234 @@
+            {
+                var completionChange = value switch
+                {
+                    CompletionHandlerConversion => await HandleConversionChangeAsync(document, item, cancellationToken).ConfigureAwait(false),
+                    CompletionHandlerIndexer => await HandleIndexerChangeAsync(document, item, cancellationToken).ConfigureAwait(false),
+                    _ => null,
+                };
+                if (completionChange is { })
+                {
+                    return completionChange;
+                }
+            }
+
+            return await base.GetChangeAsync(document, item, completionListSpan, commitKey, disallowAddingImports, cancellationToken).ConfigureAwait(false);
+        }
+
+        private static async Task<CompletionChange?> HandleConversionChangeAsync(Document document, CompletionItem item, CancellationToken cancellationToken)
+        {
+            var position = SymbolCompletionItem.GetContextPosition(item);
+            if (!item.Properties.TryGetValue(MinimalTypeNamePropertyName, out var typeName))
+            {
+                return null;
+            }
+
+            var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var tokenAtPosition = root.FindTokenOnLeftOfPosition(position);
+            var normalizedToken = tokenAtPosition.GetPreviousTokenIfTouchingWord(position);
+            // syntax tree manipulations are to complicated if a mixture of conditionals is involved. Some text manipulation is easier here.
+            //                      ↓               | cursor position
+            //                   ↓                  | normalizedToken (dot)
+            // white?.Black.White.Black?.White      | current user input
+            // white?.Black.White.Black?.White      | rootExpression (text manipulation starts with this)
+            //       .Black.White                   | parentExpression (needed to calculate the position to insert the closing brace)
+            //                    Black             | identifier at cursor position (gets removed, because the user typed the name of a type)
+            // |----------------------|             | part to replace (TextChange.Span), if identifier is not present: ends at rootExpression.End (after White.)
+            //                   ↑                  | insert closing brace between White and dot (parentExpression.Span.End)
+            // ((Black)white?.Black.White).?.White  | The result. Because we removed the identifier, the remainder after the identifier may be syntactically wrong 
+            //                             ↑        | cursor after the manipulation is placed after the dot
+            var rootExpression = GetRootExpressionOfInvocation(normalizedToken);
+            var parentExpression = GetParentExpressionOfInvocation(normalizedToken);
+            var identifier = tokenAtPosition.Parent as IdentifierNameSyntax;
```

Comment:
this may not be an identifier name.  consider something that is castable to to a type called `intfoo`.  and the user types `x.int`.  we'd show the conversoin to `intfoo` but `int` would not be an identifier.

Reply:
Fixed in 875bf0f4 and 8baae0dd

Reply:
Fixed in875bf0f4 and 8baae0dd.